### PR TITLE
Add a key for replacing the top layer

### PR DESCRIPTION
**Pull request by xs5871**
Reference: [chat discussion](https://kmkfw.zulipchat.com/#narrow/stream/384078-KMK-development/topic/adding.20an.20extra.20way.20of.20switching.20layers)

### CODE REVIEW

File: kmk/modules/layers.py [hunk 1]
```diff
@@ -79,11 +82,17 @@ def __init__(
...
+        Switches the top layer
+        '''
+        self.activate_layer(keyboard, key.meta.layer, idx=0)
+
     def _df_pressed(self, key, keyboard, *args, **kwargs):
         '''
         Switches the default layer
         '''
-        self.activate_layer(keyboard, key.meta.layer, as_default=True)
+        self.activate_layer(keyboard, key.meta.layer, idx=-1)
```
Comment (regicidalplutophage) [inline kmk/modules/layers.py:95]: I just really want to say I really like this!

**Review by regicidalplutophage — APPROVED** [review]
:+1:

**Comment by ronsmits** [comment]
so Using this change I expected that the base layer and the current layer are active, thet are not. It changes the default layer and it seems other layers are not active. Running it from thonny shows that indeed all other layers are deactivated:
```
200175719 kmk.keyboard: coordkeys_pressed={0: Key(code=1010, has_modifiers=None)}
200175725 kmk.modules.layers: active_layers=[1]
200175849 kmk.keyboard: <Event: key_number 0 released>: Key(code=1010, has_modifiers=None)
200175855 kmk.keyboard: coordkeys_pressed={}
200178848 kmk.keyboard: <Event: key_number 0 pressed>: Key(code=1011, has_modifiers=None)
200178854 kmk.keyboard: coordkeys_pressed={0: Key(code=1011, has_modifiers=None)}
200178860 kmk.modules.layers: active_layers=[2]
200178980 kmk.keyboard: <Event: key_number 0 released>: Key(code=1011, has_modifiers=None)
200178986 kmk.keyboard: coordkeys_pressed={}
200180793 kmk.keyboard: <Event: key_number 0 pressed>: Key(code=1012, has_modifiers=None)
200180797 kmk.keyboard: coordkeys_pressed={0: Key(code=1012, has_modifiers=None)}
200180803 kmk.modules.layers: active_layers=[0]
200180924 kmk.keyboard: <Event: key_number 0 released>: Key(code=1012, has_modifiers=None)
200180931 kmk.keyboard: coordkeys_pressed={}
```

**Comment by xs5871** [comment]
`KC.FD` replaces the top most layer. If only the "base" layer (i.e. lowest layer) is active, then that layer is also the top layer -- in that case `KC.DF(...)` and `KC.FD(...)` have identical behavior. Btw just to clarify, there's no such thing as a default or current layer, those are colloquialisms; there's a list of active layers with key positions being resolved by layer order and key transparency.

I'm guessing you probably want to keep layer 0 as the lowest layer. If so, you want to add another layer on top first, and then switch out that top layer, then activating only layer 0 to complete the cycle. Example snippet for 1 base and 3 overlay layers:
```python
keymap = [
[KC.TG(1)],
[KC.FD(2)],
[KC.FD(3)],
[KC.TO(0)],
]
```

**Comment by ronsmits** [comment]
Thank you very much for explaining this, it works now :)

**Pull request merged**